### PR TITLE
Update the default module requirements from python-2.6/boto to python-3.6/botocore-1.16 (breaking change)

### PR DESCRIPTION
**Pull request by tremble**
Depends-On: https://github.com/ansible-collections/amazon.aws/pull/302

##### SUMMARY

The vast bulk of modules are now boto3 based, but some even list python2.6/boto as their requirements because they've picked the default from the aws docs fragment.

Switch the defaults over to reduce the copy&paste and reflect that

- boto v2 is deprecated
- python 2.6 is no longer supported by the AWS collections

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

plugins/doc_fragments/aws.py

##### ADDITIONAL INFORMATION

Should wait for the 2.0 release so we can cleanly update everything.

### CODE REVIEW

**Comment by tremble** [comment]
@jillr in your presentation I think you suggested dropping 2.7 support as a part of the 2.0.0 release since boto3 is going to drop support soon.  If we're going to do that I guess we update the python requirement in here to 3.6 rather than 2.7.

**Comment by goneri** [comment]
recheck

**Comment by jillr** [comment]
@tremble Yeah 1.5 will be the last 2.x supporting release, so we can bump this to py3.6 for collection version 2.0

**Review by jillr — APPROVED** [review]
I'm very excited about this PR. :)  LGTM once the merge conflict is resolved.
@tremble do you have a PR queued up to add the boto requirement to community modules that are still boto2 or shall I take care of that?

**Comment by tremble** [comment]
@jillr I don't currently have a PR lined up for community.aws but I can get that lined up in the next couple of days, was mostly waiting on this to land (and decisions from your side on version support policies)

I've also added botocore >= 1.16.0 based on the policy in https://github.com/ansible/ansible/pull/74585 (in theory we could bump this to something around 1.16.5 or even 1.17.0 but now there's a general policy outlined I'd be inclined to go slightly older (and with .0 release) knowing we can bump it with the next 'major' release.

**Comment by jillr** [comment]
recheck

**Pull request merged**
